webgpu.h

```diff
@@ -0,0 +1,776 @@
+    WGPUTextureView resolveTarget;
+    WGPULoadOp loadOp;
+    WGPUStoreOp storeOp;
+    WGPUColor clearColor;
+} WGPURenderPassColorAttachmentDescriptor;
+
+typedef struct WGPUTextureCopyView {
+    void const * nextInChain;
+    WGPUTexture texture;
+    uint32_t mipLevel;
+    uint32_t arrayLayer;
+    WGPUOrigin3D origin;
+} WGPUTextureCopyView;
+
+typedef struct WGPUTextureDescriptor {
+    void const * nextInChain;
+    WGPUTextureUsageFlags usage;
+    WGPUTextureDimension dimension;
+    WGPUExtent3D size;
+    uint32_t arrayLayerCount;
+    WGPUTextureFormat format;
+    uint32_t mipLevelCount;
+    uint32_t sampleCount;
+} WGPUTextureDescriptor;
+
+typedef struct WGPUVertexBufferDescriptor {
+    uint64_t stride;
+    WGPUInputStepMode stepMode;
+    uint32_t attributeCount;
+    WGPUVertexAttributeDescriptor const * attributes;
+} WGPUVertexBufferDescriptor;
+
+typedef struct WGPURenderPassDescriptor {
+    uint32_t colorAttachmentCount;
+    WGPURenderPassColorAttachmentDescriptor const * colorAttachments;
+    WGPURenderPassDepthStencilAttachmentDescriptor const * depthStencilAttachment;
+} WGPURenderPassDescriptor;
+
+typedef struct WGPUVertexInputDescriptor {
+    void const * nextInChain;
+    WGPUIndexFormat indexFormat;
+    uint32_t bufferCount;
+    WGPUVertexBufferDescriptor const * buffers;
+} WGPUVertexInputDescriptor;
+
+typedef struct WGPURenderPipelineDescriptor {
+    void const * nextInChain;
+    WGPUPipelineLayout layout;
+    WGPUPipelineStageDescriptor vertexStage;
+    WGPUPipelineStageDescriptor const * fragmentStage;
+    WGPUVertexInputDescriptor vertexInput;
+    WGPUPrimitiveTopology primitiveTopology;
+    WGPURasterizationStateDescriptor const * rasterizationState;
+    uint32_t sampleCount;
+    WGPUDepthStencilStateDescriptor const * depthStencilState;
+    uint32_t colorStateCount;
+    WGPUColorStateDescriptor const * colorStates;
+    uint32_t sampleMask;
+    bool alphaToCoverageEnabled;
+} WGPURenderPipelineDescriptor;
+
+
+typedef void (*WGPUBufferCreateMappedCallback)(WGPUBufferMapAsyncStatus status,
+                                               WGPUCreateBufferMappedResult result,
+                                               void* userdata);
+typedef void (*WGPUBufferMapReadCallback)(WGPUBufferMapAsyncStatus status,
+                                          const void* data,
+                                          uint64_t dataLength,
+                                          void* userdata);
+typedef void (*WGPUBufferMapWriteCallback)(WGPUBufferMapAsyncStatus status,
+                                           void* data,
+                                           uint64_t dataLength,
+                                           void* userdata);
+typedef void (*WGPUFenceOnCompletionCallback)(WGPUFenceCompletionStatus status, void* userdata);
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Procs of Buffer
+typedef void (*WGPUProcBufferUnmap)(WGPUBuffer buffer);
```

Comment:
Since there is no ability to get a proc address, these typedefs aren't useful. Let's not add them right now?

Reply:
Removed (for now)